src/agentNet/agent_net.py

```diff
@@ -0,0 +1,83 @@
+import networkx as nx
+import random
+import matplotlib.pyplot as plt
+import yaml
+from network_utils import *
+
+class AgentNet(nx.Graph):
+    """
+    Class with agent network graph.
+    """
+    def __init__(self, init_num=2, net_type="complete", net_density="MEDIUM"):
+        """
+        Method initializes an agent network with default parameters.
+        :param init_num: initial number of nodes for random graph construction algorithms or initial nodes for complete graph
+        :param net_type: type of graph {"complete", "random", "custom"}
+        :param net_density: density of random graph {SPARSE, MEDIUM, DENSE}
+        """
+        self.init_num =  init_num 
+        self.net_type = net_type 
+        self.net_density = net_density 
+    def create_network(self):
+        """
+        Method creates an agent network up to net_type.
+        """
+        if self.net_type == "complete":
+            self.network = nx.complete_graph(n = self.init_num)
+        elif self.net_type == "custom":
+            self.network = nx.Graph(self.connections)
+        else:
+            self.network = nx.fast_gnp_random_graph(self.init_num, dens2prob(self.net_density))
+    def set_params(self, config_filename):
```

Comment:
z tego co widzę to to ustawia losowe polityki na bazie jakiegoś konfiga - proponuję zmienić nazwę aby było widać co tak właściwie się dzieje i dodać metodę na ustawianie konfiguracji agentów deterministycznie